test/xunit.analyzers.tests/TestClassMustBePublicTests.cs

```diff
@@ -1,25 +1,142 @@
-﻿using Microsoft.CodeAnalysis.Diagnostics;
+﻿using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Diagnostics;
 
 namespace Xunit.Analyzers
 {
     public class TestClassMustBePublicTests
     {
-        readonly DiagnosticAnalyzer analyzer = new TestClassMustBePublic();
+        private readonly DiagnosticAnalyzer analyzer = new TestClassMustBePublic();
+
+        private static IEnumerable<object[]> CreateFactsInNonPublicClassCases()
+        {
+            foreach (var factAttribute in new[] {"Xunit.Fact", "Xunit.Theory"})
+            {
+                foreach (var nonPublicAccessModifierForClass in new[] {"", "internal"})
+                {
+                    yield return new object[] {factAttribute, nonPublicAccessModifierForClass};
+                }
+            }
+        }
 
         [Fact]
-        public async void DoesNotFindErrorForPublicClass()
+        public async void ForPublicClass_DoesNotFindError()
+        {
+            var source = "public class TestClass { [Xunit.Fact] public void TestMethod() { } }";
+
+            var diagnostics = await CodeAnalyzerHelper.GetDiagnosticsAsync(analyzer, source);
+
+            Assert.Empty(diagnostics);
+        }
+
+        [Theory]
+        [MemberData(nameof(CreateFactsInNonPublicClassCases))]
+        public async void ForFriendOrInternalClass_FindsError(
+            string factRelatedAttribute,
+            string classAccessModifier)
```

Comment:
The `classAccessModifier` parameter does not seem to be used.

Reply:
Overlooked. Fixed now, thx